src/charm.py

```diff
@@ -112,15 +131,41 @@ def _configure(self, _):
             reloaded = self._prometheus_server.reload_configuration()
             if not reloaded:
                 self.unit.status = BlockedStatus("Failed to load Prometheus config")
+                return
             else:
                 self.unit.status = ActiveStatus()
                 logger.info("Prometheus configuration reloaded")
         else:
             container.add_layer(self._name, new_layer, combine=True)
             container.restart(self._name)
             logger.info("Prometheus (re)started")
+
+        remote_write_plus_ingress_status_message = INGRESS_MULTIPLE_UNITS_STATUS_MESSAGE
+
+        if not self._validate_ingress_and_remote_write():
+            self.unit.status = BlockedStatus(remote_write_plus_ingress_status_message)
+            logger.error(
+                "Using the ingress relation and receiving remote_write relations with more than "
+                "one Prometheus unit; remote_write data is likely to be sent to only one unit."
+            )
+        elif (
+            isinstance(self.unit.status, BlockedStatus)
+            and self.unit.status.message == remote_write_plus_ingress_status_message
+        ):
+            self.unit.status = ActiveStatus()
+
+        # Ensure the right address is set on the remote_write relations
+        self.remote_write_provider.update_endpoint()
+
+        if not isinstance(self.unit.status, BlockedStatus):
             self.unit.status = ActiveStatus()
 
+    def _validate_ingress_and_remote_write(self) -> bool:
+        if not (peer_relation := self.model.get_relation("prometheus-peers")):
+            return True
+
+        return not (self.model.get_relation("ingress") and len(peer_relation.units) > 0)
```

Comment:
Seems like this check could be handy in other charms.
That being said, iirc, this charm implicitly assumes in several places there is only one unit.